### PR TITLE
JavaScript-JavaScript3-Week3

### PR DESCRIPTION
**Pull request by Divyajg**
Javascript files with promises and classes.

### CODE REVIEW

File: javascript/javascript3/week3/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        context.stroke();
+    }
+}
+
+const c1 = new Circle(90, 80, 50, 0, 2 * Math.PI, "#0008e3");
+c1.draw();
+
+
+function fillingColor() {
+    return 'rgb(' + Math.floor(Math.random() * 255) + "," + Math.floor(Math.random() * 255) + "," + Math.floor(Math.random() * 255) + ')';
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/index.js:27]: Yes, a function most people should have created!
I would call it `getRandomColor()`

File: javascript/javascript3/week3/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+function createCircle() {
+    const canvas = document.getElementById('myCanvas')
+    canvas.width = window.innerWidth
+    canvas.height = window.innerHeight
+    const circleWidth = Math.floor(Math.random() * window.screen.width);
+    const circleHeight = Math.floor(Math.random() * window.screen.height);
+    const circleRadius = Math.floor(Math.random() * 100);
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/index.js:36]: introduce `getRandomInt()` and reduce repetitive code

File: javascript/javascript3/week3/shopping-cart.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    convertToCurrency(currency) {
+        fetch(
+                "https://v6.exchangerate-api.com/v6/2d116b855d76dabb3c7dcbe8/latest/DKK"
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/shopping-cart.js:9]: why is it DKK? This is not generic.

File: javascript/javascript3/week3/shopping-cart.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    searchProduct(productName) {
+        const searchedProducts = this.products.filter((product) =>
+            product.name.toLowerCase().includes(productName.toLowerCase())
+        );
+        return searchedProducts;
+    }
+
+    getTotal() {
+        return this.products.reduce((totalPrice, product) => totalPrice + product.price, 0);
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/shopping-cart.js:43]: yes

File: javascript/javascript3/week3/shopping-cart.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            price.innerHTML = `Price: ${product.price}`;
+
+            const total = document.getElementById("total");
+            total.innerHTML = `Total Price: ${this.getTotal()}`;
+
+        });
+    }
+
+    getUser() {
+        return (fetch("https://jsonplaceholder.typicode.com/users/1")
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/shopping-cart.js:67]: what is jsonplaceholder? Which user are we getting? always `1`?

File: javascript/javascript3/week3/shopping-cart.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+const shoppingCart = new ShoppingCart();
+
+const flatscreen = new Product("flat-screen", 5000);
+shoppingCart.addProduct(flatscreen);
+flatscreen.convertToCurrency("EUR");
+
+const jabra = new Product("jabra", 700);
+shoppingCart.addProduct(jabra);
+jabra.convertToCurrency("EUR");
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/shopping-cart.js:91]: it will always get DKK

**Review by aleks-hyf — COMMENTED** [review]
A few important comments here, your functions are not generic, but hard coded often. Please fix

File: javascript/javascript3/week3/gitHub-repos.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+const url3 = fetch(
+    `https://api.github.com/search/repositories?q=user:suvarna-ratna`
+).then((response) => response.json());
+
+Promise.all([url1, url2, url3])
+    .then((users) => {
+        console.log(users);
+        users.forEach((user) => {
+            user.items.forEach((item) => {
+                const user = document.createElement("div");
```
[the remark this answers is not in the document]
Comment (Divyajg) [reply javascript/javascript3/week3/gitHub-repos.js:16]: got it.

File: javascript/javascript3/week3/index.js (hunk 1 above)
Comment (Divyajg) [reply javascript/javascript3/week3/index.js:27]: okay

File: javascript/javascript3/week3/shopping-cart.js (hunk 3 above)
Comment (Divyajg) [reply javascript/javascript3/week3/shopping-cart.js:9]: I have no knowledge of generic.  But in that api, the price given is taken as dkk by default. please correct me, what I am wrong.

File: javascript/javascript3/week3/shopping-cart.js (hunk 5 above)
Comment (Divyajg) [reply javascript/javascript3/week3/shopping-cart.js:67]: That was the api given in the hyf-homework page to use.

File: javascript/javascript3/week3/shopping-cart.js (hunk 6 above)
Comment (Divyajg) [reply javascript/javascript3/week3/shopping-cart.js:91]: Its giving correct value, but I am not sure how to render it, in the constructor of class Product.

File: javascript/javascript3/week3/shopping-cart.js (hunk 3 above)
Comment (aleks-hyf) [reply javascript/javascript3/week3/shopping-cart.js:9]: You are not using the `currency` variable. So what is this converting to?

File: javascript/javascript3/week3/shopping-cart.js (hunk 5 above)
Comment (aleks-hyf) [reply javascript/javascript3/week3/shopping-cart.js:67]: Yes the `https://jsonplaceholder.typicode.com` is the API, but this call will always get user number 1. So to be more specific, you should be passing in a function argument like so:
`getUser(user)` and then using the variable `user` inside the fetch. Hope that's clearer.

File: javascript/javascript3/week3/shopping-cart.js (hunk 6 above)
Comment (aleks-hyf) [reply javascript/javascript3/week3/shopping-cart.js:91]: Where is it giving the correct value? How do you know this? :) There is no console.log or a return value.

File: javascript/javascript3/week3/shopping-cart.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    convertToCurrency(currency) {
+        fetch(
+                "https://v6.exchangerate-api.com/v6/2d116b855d76dabb3c7dcbe8/latest/DKK"
+            )
+            .then((response) => response.json())
+            .then((data) => {
+                const currencies = data.conversion_rates;
+                const convertCurrency = this.price * currencies[currency];
```
Comment (aleks-hyf) [inline javascript/javascript3/week3/shopping-cart.js:14]: My fault, you're using currency here!

**Pull request merged**